### DISCUSSION TITLE
Primehub imu.heading()

### DISCUSSION THREAD

**Discussion by ws1088** [pybricks/support#242] [Q&A] [answered]
Hi All,
I just wonder why `imu.heading()` is not yet implemented, when `imu.tilt()` is already available? I would like to learn what is the implementation difference on the 3 different axis in the gryoscope? 
Thanks!

**Comment by laurensvalk** [comment]
Tilt can be computed directly from the accelerometer data. The heading needs an algorithm to integrate the gyro rate data over time. It will come, but we have a few more pressing features to do first 😄.

**Reply by ws1088** [reply]
so if i want to use the "heading" now, can I rotate my hub so that the side of the hub is pointing upward, and then use the tilt() data?!

**Reply by laurensvalk** [reply]
No. Tilt is computed using gravity, which does not give you a heading.

**Comment by laurensvalk** [comment, marked as answer]
A new answer can be found in https://github.com/pybricks/support/issues/989, so we can close this thread.
